brave/src/main/java/brave/Tracer.java

```diff
@@ -129,22 +148,65 @@ public Tracer build() {
   final Reporter<zipkin2.Span> reporter; // for toString
   final Recorder recorder;
   final Sampler sampler;
+  final ErrorParser errorParser;
   final CurrentTraceContext currentTraceContext;
   final boolean traceId128Bit, supportsJoin;
   final AtomicBoolean noop;
-  final ErrorParser errorParser;
 
-  Tracer(Tracing.Builder builder, Clock clock, AtomicBoolean noop, ErrorParser errorParser) {
-    this.noop = noop;
-    this.propagationFactory = builder.propagationFactory;
-    this.supportsJoin = builder.supportsJoin && propagationFactory.supportsJoin();
+  Tracer(
+      Clock clock,
+      Propagation.Factory propagationFactory,
+      Reporter<zipkin2.Span> reporter,
+      Recorder recorder,
+      Sampler sampler,
+      ErrorParser errorParser,
+      CurrentTraceContext currentTraceContext,
+      boolean traceId128Bit,
+      boolean supportsJoin,
+      AtomicBoolean noop
+  ) {
     this.clock = clock;
-    this.reporter = builder.reporter;
-    this.recorder = new Recorder(builder.endpoint, clock, builder.reporter, this.noop);
-    this.sampler = builder.sampler;
-    this.currentTraceContext = builder.currentTraceContext;
-    this.traceId128Bit = builder.traceId128Bit || propagationFactory.requires128BitTraceId();
+    this.propagationFactory = propagationFactory;
+    this.reporter = reporter;
+    this.recorder = recorder;
+    this.sampler = sampler;
     this.errorParser = errorParser;
+    this.currentTraceContext = currentTraceContext;
+    this.traceId128Bit = traceId128Bit;
+    this.supportsJoin = supportsJoin;
+    this.noop = noop;
+  }
+
+  /**
+   * This to temporarily override the sampler used when starting new local traces. This is only used
+   * in advanced scenarios like looking at annotations to make a decision.
+   *
+   * <p>Example:
+   * <pre>{@code
+   * Sampler declarativeOverride = declarativeSampler.toSampler(method, tracing.sampler());
+   * Tracer tracer = tracing.tracer().withSampler(declarativeOverride);
+   *
+   * // Now, commands like startScopedSpan use a method-specific decision when starting a new trace
+   * }</pre>
+   *
+   * <p>See {@code brave.http.HttpSampler.toSampler()}
+   *
+   * @see DeclarativeSampler
+   */
+  public Tracer withSampler(Sampler sampler) {
```

Comment:
note: this hook is needed in order to permit ease of use when using aspectj or other declarative samplers for local originated traces